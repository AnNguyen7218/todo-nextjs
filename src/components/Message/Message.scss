@import 'config.gutters';
@import 'config.fonts';
@import 'config.colors';

.container {
  padding: gutter(8) gutter(16);
  width: 100%;

  &--error {
    background-color: $light-red;
    border: 1px solid $red-border;
  }
}

.content {
  @include small;

  &--error {
    color: $red;
  }
}
