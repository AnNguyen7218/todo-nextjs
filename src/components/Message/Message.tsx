import React from 'react';
import cx from 'classnames';
import css from './Message.scss';

type Props = {
  children: React.ReactChild;
  theme?: 'error';
  isShow: boolean;
  className?: string;
  contentClassName?: string;
};

const Message: React.FC<Props> = ({
  children,
  theme = 'error',
  isShow,
  className,
  contentClassName,
}) => {
  if (!isShow) {
    return null;
  }

  return (
    <div
      className={cx(css['container'], css[`container--${theme}`], className)}
    >
      <div
        className={cx(
          css['content-container'],
          css[`content--${theme}`],
          contentClassName
        )}
      >
        {children}
      </div>
    </div>
  );
};

export default Message;
