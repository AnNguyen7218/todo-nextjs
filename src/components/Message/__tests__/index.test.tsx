import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Message from '../Message';

afterEach(cleanup);

describe('Message Component', () => {
  test('it should match snapshot', () => {
    const { container } = render(
      <>
        <Message isShow theme="error">
          Error show message
        </Message>

        <Message isShow={false}>Error hide</Message>
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
