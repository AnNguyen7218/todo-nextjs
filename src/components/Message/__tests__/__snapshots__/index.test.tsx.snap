// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component it should match snapshot 1`] = `
<div>
  <div
    class="container container--error"
  >
    <div
      class="content-container content--error"
    >
      Error show message
    </div>
  </div>
</div>
`;
